src/Facebook/FileUpload/FacebookResumableUploader.php

```diff
@@ -0,0 +1,208 @@
+
+    /**
+     * Upload - phase start
+     *
+     * @param $endpoint
+     * @param $filePath
+     * @return FacebookTransferChunk
+     *
+     * @throws FacebookUploadException
+     * @throws FacebookSDKException
+     */
+    public function start($endpoint, $filePath)
+    {
+        $fileSize = filesize($filePath);
+
+        $startReqParams = [
+            'upload_phase' => 'start',
+            'file_size' => $fileSize,
+        ];
+
+        $request = new FacebookRequest(
+            $this->app,
+            $this->accessToken,
+            'POST',
+            $endpoint,
+            $startReqParams
+        );
+
+        try {
+            $res= $this->client->sendRequest($request)->getDecodedBody();
+
+            $firstTransferChunk = new FacebookTransferChunk(
+                $filePath,
+                $res['upload_session_id'],
+                $res['video_id'],
+                $res['start_offset'],
+                $res['end_offset'] - $res['start_offset']
+            );
+
+            return $firstTransferChunk;
+        } catch (FacebookResponseException $e) {
+            throw new FacebookUploadException($e->getResponse());
+        }
+    }
+
+    /**
+     * Upload - phase transfer
+     *
+     * @param $endpoint
+     * @param FacebookTransferChunk $chunk
+     * @return FacebookTransferChunk
+     *
+     * @throws FacebookUploadTransferException
+     * @throws FacebookSDKException
+     */
+    public function transfer($endpoint, FacebookTransferChunk $chunk)
+    {
+        $transReqParams = [
+            'upload_phase' => 'transfer',
+            'upload_session_id' => $chunk->getUploadSessionId(),
+            'start_offset' => $chunk->getStartOffset(),
+            'video_file_chunk' => $chunk,
+        ];
+
+        $maxTransferTries = $this->maxTransferTries;
+
+        $request = new FacebookRequest(
+            $this->app,
+            $this->accessToken,
+            'POST',
+            $endpoint,
+            $transReqParams
+        );
+
+        while ($maxTransferTries > 0) {
+            try {
+                $res= $this->client->sendRequest($request)->getDecodedBody();
+
+                $nextTransferChunk = new FacebookTransferChunk(
+                    $chunk->getFilePath(),
+                    $chunk->getUploadSessionId(),
+                    $chunk->getVideoId(),
+                    $res['start_offset'],
+                    $res['end_offset'] - $res['start_offset']
+                );
+
+                return $nextTransferChunk;
+            } catch (FacebookResponseException $e) {
+                if (--$maxTransferTries <= 0) {
+                    $resumeContext = new FacebookResumeContext(
+                        $endpoint,
+                        $this->accessToken,
+                        $chunk
+                    );
+                    throw new FacebookUploadTransferException($e->getResponse(), $resumeContext);
+                }
+                usleep(500000);
```

Comment:
Is this 100 necessary? If so, let's come up with a different method than `usleep()`. This will keep all the resources hogged while it's waiting. Otherwise let's just kill this line.